src/EnsembleKalmanProcess.jl

```diff
@@ -486,8 +555,18 @@ The error is stored within the `EnsembleKalmanProcess`.
 """
 function compute_error!(ekp::EnsembleKalmanProcess)
     mean_g = dropdims(mean(get_g_final(ekp), dims = 2), dims = 2)
-    diff = ekp.obs_mean - mean_g
-    X = ekp.obs_noise_cov \ diff # diff: column vector
+    diff = get_obs(ekp) - mean_g
+    #   X = get_obs_noise_cov(ekp) \ diff # diff: column vector
+
+    Γ_inv = get_obs_noise_cov_inv(ekp, build = false)
+    γ_sizes = [size(γ_inv, 1) for γ_inv in Γ_inv]
+    X = zeros(sum(γ_sizes), size(diff, 2)) # stores Y' * Γ_inv
+    shift = [0]
```

Comment:
Why does this need to be an array?

Reply:
It used to be that Julia would not allow mutation in a loop, maybe this changed recently...
https://discourse.julialang.org/t/loop-scope-works-differently-for-int-and-arrays-why/28172/7

Reply:
it's a pattern that is elsewhere in this code so perhaps we can clean this up another time altogether